families/supplychain/python/sawtooth_supplychain/rest_api/sc_rest_api.py

```diff
@@ -0,0 +1,190 @@
+                        action='count',
+                        default=0,
+                        help='Increase level of output sent to stderr')
+
+    return parser.parse_args(args)
+
+
+async def access_logger(app, handler):
+    """Simple logging middleware to report info about each request/response.
+    """
+    async def logging_handler(request):
+        start_time = time.time()
+        request_name = hex(int(start_time * 10000))[-6:]
+        client_ip, _ = request.transport.get_extra_info(
+            'peername', ('UNKNOWN', None))
+
+        # log request
+        LOGGER.info(
+            'Request  %s: "%s %s" from %s',
+            request_name,
+            request.method,
+            request.rel_url,
+            client_ip)
+
+        def log_response(response):
+            # pylint: disable=protected-access
+            content_length = response._headers.get('Content-Length',
+                                                   'UNKNOWN')
+            if content_length == 'UNKNOWN':
+                LOGGER.info(
+                    'Response %s: %s status, %s size, in %.3fs',
+                    request_name,
+                    response._status,
+                    content_length,
+                    time.time() - start_time)
+            else:
+                LOGGER.info(
+                    'Response %s: %s status, %sB size, in %.3fs',
+                    request_name,
+                    response._status,
+                    content_length,
+                    time.time() - start_time)
+
+        try:
+            response = await handler(request)
+            log_response(response)
+            return response
+        except web.HTTPError as e:
+            log_response(e)
+            raise e
+
+    return logging_handler
+
+
+async def cors_handler(request):
+    headers = {}
+    RouteHandler.add_cors_headers(request, headers)
+    return web.Response(headers=headers)
+
+
+def start_rest_api(host, port, stream, timeout):
+    """Builds the web app, adds route handlers, and finally starts the app.
+    """
+    loop = asyncio.get_event_loop()
+    app = web.Application(loop=loop, middlewares=[access_logger])
+
+    # Add routes to the web app
+    LOGGER.info('Creating handlers for validator at %s', stream.url)
+    handler = RouteHandler(loop, stream, timeout)
+
+    app.router.add_route('OPTIONS', '/{route_name}', cors_handler)
+
+    app.router.add_get('/agents', handler.list_agents)
+    app.router.add_get('/agents/{agent_id}', handler.fetch_agent)
+    app.router.add_get('/applications', handler.list_applications)
+    app.router.add_get('/records', handler.list_records)
+    app.router.add_get('/records/{record_id}', handler.fetch_record)
+    app.router.add_get('/records/{record_id}/applications',
+                       handler.fetch_applications)
+
+    # Start app
+    LOGGER.info('Starting Supplychain REST API on "%s:%s', host, port)
+    web.run_app(app, host=host, port=port, access_log=None)
```

Comment:
To replace the removed middleware, use aiohttp's built in logger. Replace line 137 with this:
```
    web.run_app(app, host=host, port=port,
                access_log=LOGGER,
                access_log_format='%r: %s status, %b size, in %Tf s')
```